cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_impl.i

```diff
@@ -0,0 +1,309 @@
+{
+const size_t classBufSize = 10;
+
+template <typename FPType, CpuType cpu>
+class BruteForceNearestNeighbors
+{
+public:
+    BruteForceNearestNeighbors(char metric) : _metric(metric) {}
+
+    ~BruteForceNearestNeighbors()
+    {
+        if (_distances)
+        {
+            services::internal::service_free<FPType, cpu>(_distances);
+        }
+    }
+
+    services::Status kNearest(const size_t k, int * neighborsIndices, FPType * neighborsDistances, const NumericTable * trainTable,
+                              const NumericTable * testTable)
+    {
+        computeDistances(trainTable, testTable);
+
+        const size_t nTrain = trainTable->getNumberOfRows();
+        const size_t nTest  = testTable->getNumberOfRows();
+
+        const size_t outerBlockSize = 32;
+        const size_t nOuterBlocks   = nTest / outerBlockSize + !!(nTest % outerBlockSize);
+
+        daal::threader_for(nOuterBlocks, nOuterBlocks, [&](size_t outerBlock) {
+            const size_t outerStart = outerBlock * outerBlockSize;
+            const size_t outerEnd   = outerBlock + 1 == nOuterBlocks ? nTest : outerStart + outerBlockSize;
+
+            for (size_t i = outerStart; i < outerEnd; ++i)
+            {
+                daal::services::internal::TArray<int, cpu> indicesArr(nTrain);
+                int * indices = indicesArr.get();
+                for (size_t j = 0; j < nTrain; ++j)
+                {
+                    indices[j] = j;
+                }
+
+                daal::algorithms::internal::qSort<FPType, int, cpu>(nTrain, _distances + i * nTrain, indices);
+
+                for (size_t j = 0; j < k; ++j)
+                {
+                    neighborsIndices[i * k + j]   = indices[j];
+                    neighborsDistances[i * k + j] = _distances[i * nTrain + j];
+                }
+            }
+        });
+
+        return services::Status();
+    }
+
+    services::Status kClassification(const size_t k, const size_t nClasses, VoteWeights voteWeights, const DAAL_UINT64 resultsToCompute,
+                                     const NumericTable * trainTable, const NumericTable * testTable, const NumericTable * trainLabelTable,
+                                     NumericTable * testLabelTable, NumericTable * indicesTable, NumericTable * distancesTable)
+    {
+        daal::SafeStatus s;
+
+        const size_t nTrain = trainTable->getNumberOfRows();
+        const size_t nTest  = testTable->getNumberOfRows();
+
+        daal::internal::WriteRows<FPType, cpu> distancesRows(distancesTable, 0, nTest);
+        daal::internal::WriteRows<int, cpu> indicesRows(indicesTable, 0, nTest);
+        FPType * neighborsDistances = distancesRows.get();
+        int * neighborsIndices      = indicesRows.get();
+        DAAL_CHECK_MALLOC(neighborsDistances);
+        DAAL_CHECK_MALLOC(neighborsIndices);
+
+        kNearest(k, neighborsIndices, neighborsDistances, trainTable, testTable);
+
+        if (resultsToCompute & computeClassLabels)
```

Comment:
You should rely on resultsToEvaluate here

Reply:
Fixed